File: examples/07_render_texture/script/main.lua
local texture = sn.Texture()
texture:FillColor(sn.Color(1, 1, 1, 1))
local outTexture = sn.Texture()
outTexture:Load("logo.png")
local renderTexture = sn.RenderTexture()
local s = outTexture:Size()
renderTexture:Create(s.x, s.y)

local model = sn.Model()
model:Load("Suzanne.gltf")
local draw3d = sn.Draw3D(texture)
draw3d.position = sn.Vec3(0)
draw3d.model = model

local draw2d = sn.Draw2D(outTexture)
draw2d.scale = outTexture:Size()

local pos = sn.Vec3(1, 1, 3)

function Update()
    sn.Scene.GetCamera():LookAt(pos, sn.Vec3(0), sn.Vec3(0, 1, 0))
end

function Draw()
    sn.Graphics.BeginTarget2D(renderTexture)
    -- Draw texture
    draw3d:Draw()
    sn.Graphics.EndTarget(renderTexture, outTexture)

    -- Draw render texture
    draw2d:Draw()
end

File: examples/07_render_texture/script/main.py
from sinen import *

texture = Texture()
texture.fill_color(Color(1, 1, 1, 1))
outTexture = Texture()
outTexture.load("logo.png")
render_texture = RenderTexture()
s = outTexture.size()
render_texture.create(int(s.x), int(s.y))

model = Model()
model.load("Suzanne.gltf")
draw3d = Draw3D()
draw3d.material.append(texture)
draw3d.position = Vec3(0, 0, 0)
draw3d.model = model

draw2d = Draw2D()
draw2d.material.append(outTexture)
draw2d.scale = outTexture.size()


pos = Vec3(1, 1, 3)
speed = 0.1

def update():
  if Keyboard.is_down(Keyboard.W): 
    pos.z -= speed

  if Keyboard.is_down(Keyboard.S):
    pos.z += speed

  if Keyboard.is_down(Keyboard.A):
    pos.x -= speed
  
  if Keyboard.is_down(Keyboard.D):
    pos.x += speed

  if Keyboard.is_down(Keyboard.LSHIFT):
    pos.y -= speed

  if Keyboard.is_down(Keyboard.SPACE):
    pos.y += speed

  Scene.camera().lookat(pos, Vec3(0, 0, 0), Vec3(0, 1, 0))

def draw():
  Graphics.begin_target2d(render_texture)
  # Draw texture
  draw3d.draw()
  Graphics.end_target(render_texture, outTexture)

  # Draw render texture
  draw2d.draw()
